pil-analyzer/src/evaluator.rs

```diff
@@ -655,31 +762,35 @@ mod internal {
         Ok(match b {
             BuiltinFunction::ArrayLen => match arguments.pop().unwrap().as_ref() {
                 Value::Array(arr) => Value::Integer((arr.len() as u64).into()),
-                v => Err(EvalError::TypeError(format!(
+                v => panic!(
```

Comment:
Why this change?

Reply:
Actually most of the type errors should be replaced by panic, since it's now an internal error in the type checker if this does not match up.